### PR TITLE
chore(rna): theming docs

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/connected-components/authenticator/customization/customization.styling.react-native.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+## Styling
+
+### Theme Provider
+
+You can update the style of the Authenticator by wrapping it with a [ThemeProvider](/react-native/theming). Then create a [theme object](/react-native/theming#theme-structure), with all your font customizations.
+
+```jsx file=../../../../../../../examples/react-native/src/features/Authenticator/Theming/Example.tsx
```
Comment (dbanksdesign) [inline docs/src/pages/[platform]/connected-components/authenticator/customization/customization.styling.react-native.mdx:8]: I assume this is a good idea so we can test the example even if we aren't rendering it in the docs

File: docs/src/pages/[platform]/theming/dark-mode/index.page.mdx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Dark Mode
 metaDescription: Dark Mode uses a darker color palette for all screens, making foreground content stand out against darker backgrounds.
-supportedFrameworks: react|angular|vue|flutter
+supportedFrameworks: react|angular|vue|flutter|react-native
```
Comment (ioanabrooks) [inline docs/src/pages/[platform]/theming/dark-mode/index.page.mdx:4]: nit: all

File: examples/react-native/src/features/Authenticator/Theming/DarkMode.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+import {
+  Authenticator,
+  defaultDarkModeOverride,
+  ThemeProvider,
+} from '@aws-amplify/ui-react-native';
+import { Amplify } from 'aws-amplify';
+
+import config from './aws-exports';
+import { useColorScheme } from 'react-native';
```
Comment (ioanabrooks) [inline examples/react-native/src/features/Authenticator/Theming/DarkMode.tsx:11]: nit: this shoul dbe on line 2

File: examples/react-native/src/features/Authenticator/Theming/Example.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                90: '{colors.pink.90}',
+                100: '{colors.pink.100}',
+              },
+            },
+          },
+        },
+      }}
+    >
+      <Authenticator.Provider>
+        <Authenticator></Authenticator>
```
Comment (calebpollman) [inline examples/react-native/src/features/Authenticator/Theming/Example.tsx:32]:
```suggestion
        <Authenticator />
```

File: examples/react-native/src/features/Authenticator/Theming/DarkMode.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                100: '{colors.pink.100}',
+              },
+            },
+          },
+        },
+        overrides: [defaultDarkModeOverride],
+      }}
+    >
+      <Authenticator.Provider>
+        <Authenticator></Authenticator>
```
Comment (calebpollman) [inline examples/react-native/src/features/Authenticator/Theming/DarkMode.tsx:40]:
````suggestion
        <Authenticator />
```

**Review by calebpollman — APPROVED** [review]
LGTM, left a couple suggestions in teh examples

**Pull request merged**